ui/inspectors/system-section.reel/language-and-region.reel/language-and-region.js

```diff
@@ -46,20 +107,57 @@ exports.LanguageAndRegion = Component.specialize(/** @lends LanguageAndRegion# *
                 Promise.all(loadingPromises).then(function() {
                     this.isLoading = false;
                 });
+                var today = new Date();
+                this.dateFormatShortOptions = this.generateDateFormatConvertedList(today, this.shortDateFormats);
+                this.dateFormatMediumOptions = this.generateDateFormatConvertedList(today, this.mediumDateFormats);
+                this.dateFormatLongOptions = this.generateDateFormatConvertedList(today, this.longDateFormats);
+                this.dateFormatFullOptions = this.generateDateFormatConvertedList(today, this.fullDateFormats);
+                this.timeFormatShortOptions = this.generateDateFormatConvertedList(today, this.shortTimeFormats);
+                this.timeFormatMediumOptions = this.generateDateFormatConvertedList(today, this.mediumTimeFormats);
+                this.timeFormatLongOptions = this.generateDateFormatConvertedList(today, this.longTimeFormats);
+            }
+        }
+    },
+
+    generateDateFormatConvertedList: {
+        value: function(today, dateOptionList) {
+            var formattedDateList = [];
+            for (var i = 0,length = dateOptionList.length; i < length; i++) {
+                this.dateConverter.pattern = dateOptionList[i];
+                formattedDateList.push({label: this.dateConverter.convert(today), value: dateOptionList[i]});
             }
+            return formattedDateList;
+        }
+    },
+
+    setDefaultDateAndTimeFormat: {
+        value: function(userSettings) {
+            userSettings.timeFormatShort = userSettings.timeFormatShort || this.shortTimeFormats[0];
+            userSettings.timeFormatMedium = userSettings.timeFormatMedium || this.mediumTimeFormats[0];
+            userSettings.timeFormatLong = userSettings.timeFormatLong || this.longTimeFormats[0];
+            userSettings.dateFormatShort = userSettings.dateFormatShort || this.shortDateFormats[0];
+            userSettings.dateFormatMedium = userSettings.dateFormatMedium || this.mediumDateFormats[0];
+            userSettings.dateFormatLong = userSettings.dateFormatLong || this.longDateFormats[0];
+            userSettings.dateFormatFull = userSettings.dateFormatFull || this.fullDateFormats[0];
         }
     },
 
     save: {
         value: function() {
-            return this.application.systemGeneralService.saveGeneralData(this.generalData);
+            var loadingPromises = [];
+            loadingPromises.push(
+                this.application.applicationContextService.save(),
+                this.application.systemGeneralService.saveGeneralData(this.generalData)
+            );
+            return Promise.all(loadingPromises);
```

Comment:
There is no need declaring a variable here, you should inline the array in the `Promise.all` argument.